### PR TITLE
Published Rust Bindings

### LINKED ISSUE #2730: Rust Bindings
Starting roughly in the middle of July, I would like to implement Rust bindings for Elektra. 

I think that [rust-bindgen](https://rust-lang.github.io/rust-bindgen/) should be able to automatically generate (some or all of) the bindings. I still expect there to be quite a bit of manual work to get them working properly. From my current understanding and from @kodebach's comment, this will result in a `elektra-sys` crate.
Once it is working I'll add a safe API in Rust, so that it can be used in regular Rust without the need to call unsafe code. This will then be the `elektra` crate.
I'll then make sure that they are correct by testing with cargo tests.

The typical way to document crates is with [comments in the code](https://doc.rust-lang.org/book/ch14-02-publishing-to-crates-io.html). [docs.rs](https://docs.rs) will automatically build the documentation and make it publicly available, so I think doing documentation this way makes the most sense.

To publish the crate to [crates.io](https://crates.io), an [account with an API token](https://doc.rust-lang.org/cargo/commands/cargo-login.html) is needed. As discussed with @markus2330, this account should be part of ElektraInitiative such that it is accessible to future maintainers.

I'll look at the CMake integration at the start of the project, since I'm not familiar with CMake at the moment.

Is there anything else I should add?

### PR DESCRIPTION
**Pull request by PhilippGackstatter**
The Rust Bindings are now published to crates.io. Since there are some differences between what cmake requires and what I need for publishing, I had to add modified versions of `Cargo.toml` and `build.rs`. I also added some additional documentation in the elektra crate and added a short guide for publishing future crate versions.

fixes #2730 

## Basics

Check relevant points but **please do not remove entries**.
Do not describe the purpose of this PR in the PR description but:

- [x] Short descriptions should be in the release notes (added as entry in
      `doc/news/_preparation_next_release.md` which contains `_(my name)_`)
      **Please always add something to the the release notes.**
- [ ] Longer descriptions should be in documentation or in design decisions.
- [x] Describe details of how you changed the code in commit messages
      (first line should have `module: short statement` syntax)
- [ ] References to issues, e.g. `close #X`, should be in the commit messages.

## Checklist

Check relevant points but **please do not remove entries**.
For docu fixes, spell checking, and similar none of these points below
need to be checked.

- [ ] I added unit tests for my code
- [ ] I fixed all affected documentation
- [ ] I added code comments, logging, and assertions as appropriate (see [Coding Guidelines](https://master.libelektra.org/doc/CODING.md))
- [ ] I updated all meta data (e.g. README.md of plugins and [METADATA.ini](https://master.libelektra.org/doc/METADATA.ini))
- [ ] I mentioned every code not directly written by me in [THIRD-PARTY-LICENSES](doc/THIRD-PARTY-LICENSES)

## Review

Reviewers will usually check the following:

- [ ] Documentation is introductory, concise and good to read
- [ ] Examples are well chosen and understandable
- [ ] Code is conforming to [our Coding Guidelines](https://master.libelektra.org/doc/CODING.md)
- [ ] APIs are conforming to [our Design Guidelines](https://master.libelektra.org/doc/DESIGN.md)
- [ ] Code is consistent to [our Design Decisions](https://master.libelektra.org/doc/decisions)

## Labels

- Add the "work in progress" label if you do not want the PR to be reviewed yet.
- Add the "ready to merge" label **if the build server is happy** and also you
  say that everything is ready to be merged.

### CODE REVIEW

File: doc/news/_preparation_next_release.md [hunk 1]
```diff
@@ -147,7 +147,7 @@ you up to date with the multi-language support provided by Elektra.
 
 - Add the `elektra-sys` crate which contains raw bindings to libelektra for Rust. _(Philipp Gackstatter)_
 - Add the `elektra` crate which contains safe wrapper methods for the raw bindings. The crate contains bindings for the low-level API, which means that the data types `Key` and `KeySet` can now safely be used from Rust. The Rust version of the API has been designed to take advantage of Rust's type system and to be in accordance with the memory safety of Rust. For instance, the Key has been divided into `StringKey` and `BinaryKey`, to prevent type mismatches at compile time. With the binding for `KDB`, one can take advantage of the elektra ecosystem from Rust. See the [Readme](../../src/bindings/rust/README.md) for more. _(Philipp Gackstatter)_
-- The [elektra](https://crates.io/crates/elektra) and [elektra-sys](https://crates.io/crates/elektra-sys) crates have been published to crates.io for easier usage. _(Philipp Gackstatter)_
```
Comment (markus2330) [inline doc/news/_preparation_next_release.md]: Why do you remove the links?

Comment (PhilippGackstatter) [reply doc/news/_preparation_next_release.md]: They don't pass the link checker.
Somehow it returns a NotFound, and I don't think we can fix this.

```
$ curl https://crates.io/crate/elektra
{"errors":[{"detail":"Not Found"}]}
```

Comment (markus2330) [reply doc/news/_preparation_next_release.md]: Then put the links to the whitelist. They obviously work in the browser. (But beware of typos, in the curl statement above there is a typo: crate -> crates).

File: tests/linkchecker.whitelist [hunk 2]
```diff
@@ -10,3 +10,5 @@ https://build.libelektra.org/jenkins/job/libelektra/pipeline-syntax
 # see https://github.com/ElektraInitiative/libelektra/issues/2913
 http://webdemo.libelektra.org
 https://webdemo.libelektra.org
+https://crates.io/crates/elektra
+https://crates.io/crates/elektra-sys
```
Comment (markus2330) [inline tests/linkchecker.whitelist]: Does it work without new-line at the end?

Comment (PhilippGackstatter) [reply tests/linkchecker.whitelist]: The link checker succeeded, so I guess it works.

**Comment by markus2330** [comment]
Thank you, great work!

**Pull request merged**